tests/test_http.py

```diff
@@ -266,3 +270,29 @@ def test_GET_request_chunked_no_content_length(httpserver, chunked_flag):
     assert resp.status_code == 200
     assert "Transfer-encoding" in resp.headers
     assert "Content-length" not in resp.headers
+
+
+def test_httpserver_init_failure_no_stderr_during_cleanup(tmp_path):
+    """
+    Test that, when the server encounters an error during __init__, its cleanup
+    does not raise an error (in its __del__ method), which emits a warning onto
+    stderr.
+    """
+
+    script = textwrap.dedent("""
+        from pytest_localserver import http
+        from unittest.mock import patch
+
+        with patch("pytest_localserver.http.make_server", side_effect=RuntimeError("init failure")):
+            server = http.ContentServer()
+    """)
+
+    script_file_path = os.path.join(tmp_path, "script.py")
+
+    with open(script_file_path, "w") as script_file:
+        print(script, file=script_file)
+
+    result = subprocess.run([sys.executable, script_file_path], stderr=subprocess.PIPE)
+
+    assert b"AttributeError" not in result.stderr
```

Comment:
I'd suggest adding a comment here to note why you can't meaningfully check the exit code (which is the usual way to evaluate failure of a subprocess) to tell whether the test worked.

Hopefully some future evolution of this test will be able to use the exit code in that way, but no need to get hung up on that for now.

Reply:
I added a comment. I don't mind rewording it too if you have any suggestions.

Reply:
I wouldn't mind that, but I think something like the following helps make it more explicit why we cannot use the exit code (and specifically, why the exit code doesn't tell us if the undesired `AttributeError` was thrown):

```
    # We ensure that no warning about an AttributeError is printed on stderr
    # due to an error in the server's __del__ method. This AttributeError is
    # raised during cleanup and doesn't affect the exit code of the script, so
    # we can't use the exit code to tell whether the script did its job.
```

So that would be my suggestion. Of course you can adjust it if you feel that it misses out on something important.

Reply:
I like it! Thank you.